From d33338159efbf5b6259ca2cc9020798aa75130c1 Mon Sep 17 00:00:00 2001
From: MengLi <meng.li@windriver.com>
Date: Mon, 1 Feb 2021 14:50:40 +0800
Subject: [PATCH 2/3] driver: pfe: adjust parameters for interface
 of_get_phy_mode

According to upstream commit 0c65b2b90d13("net:
of_get_phy_mode: Change API to solve int/unit warnings"),
adjust parameters for interface of_get_phy_mode.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sw/linux-pfeng/pfeng-hw-s32g.c | 2 +-
 sw/linux-pfeng/pfeng.h         | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/sw/linux-pfeng/pfeng-hw-s32g.c b/sw/linux-pfeng/pfeng-hw-s32g.c
index 61d8a22..2cc4f8d 100644
--- a/sw/linux-pfeng/pfeng-hw-s32g.c
+++ b/sw/linux-pfeng/pfeng-hw-s32g.c
@@ -369,7 +369,7 @@ static int create_config_from_dt(struct pfeng_priv *priv)
 			eth->max_speed = SPEED_2500;
 
 		/* Interface mode */
-		eth->intf_mode = of_get_phy_mode(child);
+		of_get_phy_mode(child, &eth->intf_mode);
 		if (eth->intf_mode < 0)
 			/* for non managable interface */
 			eth->intf_mode = PHY_INTERFACE_MODE_INTERNAL;
diff --git a/sw/linux-pfeng/pfeng.h b/sw/linux-pfeng/pfeng.h
index 63b5477..93469cd 100644
--- a/sw/linux-pfeng/pfeng.h
+++ b/sw/linux-pfeng/pfeng.h
@@ -75,7 +75,7 @@ struct pfeng_eth {
 	u8				*addr;
 	u8				fixed_link;
 	u32				max_speed;
-	int				intf_mode;
+	phy_interface_t			intf_mode;
 	u32				emac_id;
 	struct device_node		*dn;
 	struct clk			*tx_clk;
-- 
2.17.1

